### PR TITLE
feat(MPX-709): collection reconciliation controller

### PR DESCRIPTION
**Pull request by dmitry-saraykin**

### CODE REVIEW

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+@RestController
+class CollectionReconciliationController(
+    private val router: BlockchainRouter<CollectionService>,
+    private val enrichmentCollectionService: EnrichmentCollectionService,
+    private val collectionRepository: CollectionRepository,
+) {
+
+    @GetMapping(value = ["/reconciliation/collections"], produces = [MediaType.APPLICATION_JSON_VALUE])
+    suspend fun getCollections(
+        @RequestParam lastUpdatedFrom: Long,
```
Comment (yakischik) [inline api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt:25]: Instant?

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+        if (shortCollections.isEmpty()) {
+            return CollectionsDto()
+        }
+        val groupedIds = shortCollections.groupBy({ it.blockchain }, { it.id.collectionId })
+
+        val unionCollections = groupedIds.flatMap {
+            router.getService(it.key).getCollectionsByIds(it.value)
+        }.groupBy { it.id }
+
+        val collections = shortCollections.mapNotNull { shortCollection ->
```
Comment (yakischik) [inline api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt:43]: This one-by-one enrichment might be slow. See CollectionApiMergeService with batch enrichment. Ideally, this batch enrichment should be moved out from this class and then reused in CollectionApiMergeService/CollectionElasticService/this controller

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt (hunk 1 above)
Comment (dmitry-saraykin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt]: fixed

File: api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt (hunk 2 above)
Comment (dmitry-saraykin) [reply api/src/main/kotlin/com/rarible/protocol/union/api/controller/CollectionReconciliationController.kt]: fixed

**Review by yakischik — APPROVED** [review]

**Pull request merged**